controller/space.go

```diff
@@ -177,20 +178,20 @@ func (c *SpaceController) List(ctx *app.ListSpaceContext) error {
 	offset, limit := computePagingLimts(ctx.PageOffset, ctx.PageLimit)
 
 	return application.Transactional(c.db, func(appl application.Application) error {
-		spaces, c, err := appl.Spaces().List(ctx.Context, &offset, &limit)
-		count := int(c)
+		spaces, cnt, err := appl.Spaces().List(ctx.Context, &offset, &limit)
 		if err != nil {
 			return jsonapi.JSONErrorResponse(ctx, err)
 		}
-
-		response := app.SpaceList{
-			Links: &app.PagingLinks{},
-			Meta:  &app.SpaceListMeta{TotalCount: count},
-			Data:  ConvertSpaces(ctx.RequestData, spaces),
-		}
-		setPagingLinks(response.Links, buildAbsoluteURL(ctx.RequestData), len(spaces), offset, limit, count)
-
-		return ctx.OK(&response)
+		return ctx.ConditionalEntities(spaces, c.config.GetCacheControlSpace, func() error {
```

Comment:
We could move this whole piece outside the TX as well. To make the TX as short lived as possible.